### PR TITLE
Feature/FF-296 upload endpoint

### PR DESCRIPTION
**Pull request by open-schnick**

### CODE REVIEW

File: src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java [hunk 1]
```diff
@@ -217,190 +215,128 @@ public boolean recursivelyDeleteFileSystemEntity(FileSystemEntity parentFileSyst
...
+                        .mimeType(fileSystemUpload.getMimeType())
+                        .size(fileSystemUpload.getSize())
+                        .lastUpdated(timeStamp)
+                        .isFile(true)
+                        .createdByUserId(authenticatedUser.getUserId())
+                        .ownerIds(newOwnerIds)
+                        .visibleForUserIds(parentFileSystemEntity.getVisibleForUserIds())
+                        .visibleForGroupIds(parentFileSystemEntity.getVisibleForGroupIds())
+                        .editableForUserIds(parentFileSystemEntity.getEditableForUserIds())
+                        .editableFoGroupIds(parentFileSystemEntity.getEditableFoGroupIds())
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java:284]: Good solution

File: src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java [hunk 2]
```diff
@@ -217,190 +215,128 @@ public boolean recursivelyDeleteFileSystemEntity(FileSystemEntity parentFileSyst
...
-        long[] fileIsSharedToGroups = fileSystemEntity.getEditableFoGroupIds();
-        for (Group group : authenticatedUser.getGroups()) {
-            for (long groupId : fileIsSharedToGroups) {
-                if (groupId == group.getGroupId())
-                    return true;
+                log.info("Creating new file {} for user {}.", uploadedFile, authenticatedUser.getUserId());
+                fileSystemRepository.save(uploadedFile);
+            } else {
+                // concat pathToCheck with currentPath
+                folderPathToCheck += folderPathToCheck.equals("/") ? currentPath : "/" + currentPath;
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java:296]: Does this concat the absolute path of the parent with the relative path of the new child?

Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java:296]: yes.

File: src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java (hunk 1 above)
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java:284]: :D

File: src/main/java/de/filefighter/rest/configuration/PrepareDataBase.java [hunk 3]
```diff
@@ -278,6 +279,8 @@ private void addTestingFileSystemItems(FileSystemRepository fileSystemRepository
                         .size(4866)
                         .typeId(FOLDER.getId())
                         .visibleForGroupIds(new long[]{FAMILY.getGroupId(), ADMIN.getGroupId()})
+                        .visibleForUserIds(new long[]{0})
+                        .editableForUserIds(new long[]{0})
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/configuration/PrepareDataBase.java:283]: @open-schnick pleas move this in the new PR

File: src/main/java/de/filefighter/rest/configuration/PrepareDataBase.java [hunk 4]
```diff
@@ -289,6 +292,8 @@ private void addTestingFileSystemItems(FileSystemRepository fileSystemRepository
                         .size(0)
                         .typeId(FOLDER.getId())
                         .visibleForGroupIds(new long[]{UNDEFINED.getGroupId(), FAMILY.getGroupId(), ADMIN.getGroupId()})
+                        .visibleForUserIds(new long[]{1})
+                        .editableForUserIds(new long[]{1})
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/configuration/PrepareDataBase.java:296]: this to @open-schnick

**Comment by open-schnick** [comment]
this branch is obsolete see (#100)

**Pull request closed without merging**